### ISSUE TITLE
Is there a way to make the Trainer skip loading optimizer from a checkpoint?

### ISSUE THREAD

**Issue by nihalsid** [PyTorchLightning/pytorch-lightning#1982]
Use case is if I want to load a model's weights but reset the the optimizer state. Is there a flag I can pass that skips loading the optimizer?

**Comment by awaelchli** [comment]
There is no flag but you could implement the model hook on_load_checkpoint and reset the optimizer yourself.

**Comment by nihalsid** [comment]
Cool thanks!

**Comment by kef-bassam** [comment]
> I think you can accomplish it this way without needing to know any of the optimizer's implementation details:
> 
> ```
>     # in your lightning module...
>     def on_load_checkpoint(self, checkpoint):
>         fresh_optimizer = Adam(...)
>         self.my_optimizer.load_state_dict(fresh_optimizer.state_dict())
> ```

If I wanted to reinstantiate the optimizer and scheduler using the yaml files and the Lightning CLI, is there a best way to do that?

The way that is mentioned above hardcodes the optimizer class within the Lightning Module but I'd like to limit all of my configuration and hyperparameter tuning through a `config.yaml` that I use with the CLI. I'm not sure how to expose the contents of the configuration file to the LightningModule though.

As an example: Config before finetuning 
```
data:
  class_path: OldDataModule
# ckpt_path: 
optimizer:
  class_path: torch.optim.SGD
  init_args:
    lr: 1e-03
lr_scheduler:
  class_path: torch.optim.lr_scheduler.StepLR
  init_args:
    step_size: 2 
    gamma: 0.90 
```

Config to use for finetuning:
```
data:
  class_path: NewDataModule
ckpt_path: ./logs/pretrained.ckpt
optimizer:
  class_path: torch.optim.SGD
  init_args:
    lr: 5e-08
lr_scheduler:
  class_path: torch.optim.lr_scheduler.ExponentialLR
  init_args:
    gamma: 0.10
```

**Comment by YJ-20** [comment]
If you want to reset the optimizer state in lightning, you can use `on_load_checkpoint` as like this.
```python
def on_load_checkpoint(self, checkpoint):
    checkpoint["optimizer_states"] = []
```

**Comment by marz869** [comment]
> I think I was confused about the difference between `Trainer(resume_from_checkpoint="...")` vs. `LitModule.load_from_checkpoint("...")`, and the docs could probably be improved here.
> 
> It appears that `LightningModule.load_from_checkpoint` doesn't even instantiate your optimizers, because only the `Trainer` does that. Therefore, I believe:
> 
>     * you MUST use the Trainer's `resume_from_checkpoint` arg if you want to re-load the optimizer state (and other training state), and
> 
>     * you NEED NOT WORRY about accidentally loading other training state when calling `LightningModule.load_from_checkpoint`, because the lightning module isn't responsible for training state in the first place.
> 
> 
> Please help me out if I didn't arrive at the correct conclusion.

@colllin LightningModule.load_from_checkpoint re_load optimizer, but resume_from_checkpoint, reload epoch and global step...and reset the optimizer!! So if I have a checkpoint with all states(lr scheduler, epoch, global step, ...) if I do LightningModule.load_from_checkpoint, it re_loads optimizer, lr scheduler and optimizer works as I expect, but epoch and global step is set to 0. And if in the next line I load checkpoints again in trainer.fit, then I can continue training the model from last epoch, but optimizer is reset, and lr_scheduler does not work at all!!

**Comment by Zyriix** [comment]
A simple way is loading the checkpoint, and replacing the things you want.

Like if I want to replace the hyperparameters, I can do this:

checkpoint = torch.load(ckpt_path)
checkpoint['hyper_parameters'] = model.parameters 
torch.save(checkpoint, tmp_ckpt_path)
trainer.fit(model,tmp_ckpt_path)

I believe other things can be done in the similar way.
